core/src/main/java/org/apache/carbondata/core/datastore/compression/GzipCompressor.java

```diff
@@ -0,0 +1,138 @@
+   * @return Compressed Byte Array
+   */
+  private byte[] compressData(byte[] data) {
+    ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+    try {
+      GzipCompressorOutputStream gzipCompressorOutputStream =
+          new GzipCompressorOutputStream(byteArrayOutputStream);
+      try {
+        /**
+         * Below api will write bytes from specified byte array to the gzipCompressorOutputStream
+         * The output stream will compress the given byte array.
+         */
+        gzipCompressorOutputStream.write(data);
+      } catch (IOException e) {
+        throw new RuntimeException("Error during Compression step " + e.getMessage());
+      } finally {
+        gzipCompressorOutputStream.close();
+      }
+    } catch (IOException e) {
+      throw new RuntimeException("Error during Compression step " + e.getMessage());
+    }
+    return byteArrayOutputStream.toByteArray();
+  }
+
+  /**
+   * This method takes the Byte Array data and Deompresses in gzip format
+   *
+   * @param data   Data Byte Array for Compression
+   * @param offset Start value of Data Byte Array
+   * @param length Size of Byte Array
+   * @return
+   */
+  private byte[] decompressData(byte[] data, int offset, int length) {
+    ByteArrayInputStream byteArrayOutputStream = new ByteArrayInputStream(data, offset, length);
+    ByteArrayOutputStream byteOutputStream = new ByteArrayOutputStream();
+    try {
+      GzipCompressorInputStream gzipCompressorInputStream =
+          new GzipCompressorInputStream(byteArrayOutputStream);
+      byte[] buffer = new byte[1024];
+      int len;
+      /**
+       * Reads the next byte of the data from the input stream and stores them into buffer
+       * Data is then read from the buffer and put into byteOutputStream from a offset.
+       */
+      while ((len = gzipCompressorInputStream.read(buffer)) != -1) {
+        byteOutputStream.write(buffer, 0, len);
+      }
+    } catch (IOException e) {
+      throw new RuntimeException("Error during Decompression step " + e.getMessage());
+    }
+    return byteOutputStream.toByteArray();
+  }
+
+  @Override public byte[] compressByte(byte[] unCompInput) {
+    return compressData(unCompInput);
+  }
+
+  @Override public byte[] compressByte(byte[] unCompInput, int byteSize) {
+    return compressData(unCompInput);
+  }
+
+  @Override public byte[] unCompressByte(byte[] compInput) {
+    return decompressData(compInput, 0, compInput.length);
+  }
+
+  @Override public byte[] unCompressByte(byte[] compInput, int offset, int length) {
+    return decompressData(compInput, offset, length);
+  }
+
+  @Override public long rawUncompress(byte[] input, byte[] output) {
+    //gzip api doesnt have rawUncompress yet.
+    throw new RuntimeException("Not implemented rawUcompress for gzip yet");
+  }
+
+  @Override public long maxCompressedLength(long inputSize) {
+    if (inputSize < Integer.MAX_VALUE) {
```

Comment:
Please add some comments for this peace of code

Reply:
Done.